test/Microsoft.TestPlatform.AcceptanceTests/AcceptanceTestBase.cs

```diff
@@ -31,17 +33,22 @@ protected static void SetTestEnvironment(IntegrationTestEnvironment testEnvironm
 
         protected static string DeriveFrameworkArgValue(IntegrationTestEnvironment testEnvironment)
```

Comment:
This can be eventually re-written using a switch-case.
```
switch (testEnvironment.TargetFramework) {
case Core21TargetFramework: return Core21FrameworkArgValue;
case Core31TargetFramework: return Core31FrameworkArgValue;
case DesktopTargetFramework: return DesktopFrameworkArgValue;
default:
    throw new NotSupportedException($"{testEnvironment.TargetFramework} is not supported TargetFramework value.");
}
```
Moreover, you can use a dictionary to associate the TargetFramework with the FrameworkArgValue it should correspond to. Then search the map and if the key is not there you throw an exception, otherwise you get the value.

Reply:
Not exactly the same, switch does not do ordinal comparison but I agree that switch would be nicer. Translating to a dictionary has the same problem, and seems like unnecessary optimization. Can we keep it as is for now, and tackle this refactoring in next step where I will enable netcoreapp3.1 testing for all acceptance tests?

Reply:
My bad, the default comparison is ordinal. So the switch does ordinal. Still would rather fix it later.